### PR TITLE
React: Permission group multiselect in Add Datasets

### LINKED ISSUE #259: Add datasets: add multiselect for permission group if user has multiple
Depends on #258

### PR DESCRIPTION
**Pull request by frewmack**
Closes #259. Also changes /api/login to send the user's permission groups on login.

### CODE REVIEW

File: react/src/AddDatasets/components/DataEntryTable.tsx [hunk 1]
```diff
@@ -377,6 +387,12 @@ function DataEntryToolbar(props: {
                 <Box display="flex" flexGrow={1}>
                     <Typography variant="h6">Enter Metadata</Typography>
                 </Box>
+                <GroupDropdownSelect
+                    selectedGroupCodes={props.groups}
+                    allGroupCodes={props.allGroups}
+                    onChange={props.setGroups}
+                    disabled={props.allGroups.length <= 1}
```
Comment (kevinlul) [inline react/src/AddDatasets/components/DataEntryTable.tsx:394]: When I first sign in I always get an error on this line that `allGroups` is undefined when navigating to add datasets. This is fixed by a hard refresh.

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: react/src/AddDatasets/index.tsx [hunk 2]
```diff
@@ -28,21 +28,33 @@ const useStyles = makeStyles(theme => ({
...
     useEffect(() => {
         document.title = `Add Datasets | ${process.env.REACT_APP_NAME}`;
         handleDataChange(createEmptyRows(1)); // sets errorMessage on initial render
     }, []);
 
+    useEffect(() => {
+        if (props.groups.length > 0) {
+            setAsGroups(props.groups);
+        } else {
+            setErrorMessage("Cannot submit. You are not part of any permission groups.");
```
Comment (kevinlul) [inline react/src/AddDatasets/index.tsx:51]: Can a similar message be added for when the user attempts to submit with the field not filled out?

**Review by kevinlul — CHANGES_REQUESTED** [review]
This works great and I like that the selection is disabled if the user cannot make a meaningful choice.

File: react/src/AddDatasets/index.tsx [hunk 3]
```diff
@@ -52,6 +52,13 @@ export default function AddParticipants(props: {
         }
     }, [props.groups]);
 
+    function onChangeGroups(newGroups: string[]) {
+        setAsGroups(newGroups);
+        if (newGroups.length === 0) {
+            setErrorMessage("Cannot submit. You must select a permission group.");
+        }
```
Comment (kevinlul) [inline react/src/AddDatasets/index.tsx:59]: I think the error condition needs to consider both the filled-out state of the entire form and the dropdown. This worked initially, but when I filled out the table, the submit button was enabled again.

**Review by kevinlul — CHANGES_REQUESTED** [review]

**Review by kevinlul — APPROVED** [review]
LGTM!

**Review by annnag — CHANGES_REQUESTED** [review]
Looks good to me! Everything works as expected.
One super minor problem: when one dataset is added, the snack bar says `1 datasets...` instead of `1 dataset...`

**Review by annnag — APPROVED** [review]

**Comment by kevinlul** [comment]
We have a merge conflict because #299 was merged

**Pull request merged**